purchases/src/customEntitlementComputation/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -0,0 +1,361 @@
+package com.revenuecat.purchases
+
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import com.revenuecat.purchases.common.LogIntent
+import com.revenuecat.purchases.common.PlatformInfo
+import com.revenuecat.purchases.common.infoLog
+import com.revenuecat.purchases.common.log
+import com.revenuecat.purchases.interfaces.Callback
+import com.revenuecat.purchases.interfaces.GetStoreProductsCallback
+import com.revenuecat.purchases.interfaces.PurchaseCallback
+import com.revenuecat.purchases.interfaces.ReceiveCustomerInfoCallback
+import com.revenuecat.purchases.interfaces.ReceiveOfferingsCallback
+import com.revenuecat.purchases.interfaces.SyncPurchasesCallback
+import com.revenuecat.purchases.interfaces.UpdatedCustomerInfoListener
+import com.revenuecat.purchases.models.BillingFeature
+import com.revenuecat.purchases.models.StoreProduct
+import com.revenuecat.purchases.strings.BillingStrings
+import com.revenuecat.purchases.strings.ConfigureStrings
+import java.net.URL
+
+/**
+ * Entry point for Purchases. It should be instantiated as soon as your app has a unique user id
+ * for your user. This can be when a user logs in if you have accounts or on launch if you can
+ * generate a random user identifier.
+ * Make sure you follow the [quickstart](https://docs.revenuecat.com/docs/getting-started-1)
+ * guide to setup your RevenueCat account.
+ * @warning Only one instance of Purchases should be instantiated at a time!
+ */
+class Purchases internal constructor(
+    @get:JvmSynthetic internal val purchasesOrchestrator: PurchasesOrchestrator,
+) {
```

Comment:
I only added the functions that were needed looking at the API in iOS, but please confirm the API is what we expect